### PR TITLE
Fix SemiImplicit ball joint damping

### LINKED ISSUE #2989: Apply joint_damping to SemiImplicit ball joints
## Context

PR #2304 adds `Model.joint_damping` for always-on per-DOF damping, but review identified that `SolverSemiImplicit` still does not consume it for native `JointType.BALL` joints.

The PR is leaving this as follow-up work rather than expanding the merge-conflict/review cleanup scope.

## Current behavior

In `newton/_src/solvers/semi_implicit/kernels_body.py`, `eval_body_joints()` receives `joint_damping`, and prismatic/revolute/D6 paths pass it through `joint_force()`. The `JointType.BALL` branch only applies `joint_f`:

```python
t_total += wp.vec3(-joint_f[qd_start], -joint_f[qd_start + 1], -joint_f[qd_start + 2])
```

It never reads `joint_damping[qd_start + i]`, so `add_joint_ball()` / imported ball-joint damping has no passive damping effect in `SolverSemiImplicit`.

## Expected behavior

Ball joints should apply per-DOF passive damping consistently with the other joint types, using the ball joint angular velocity components and the three `joint_damping` entries.

## Notes

Relevant code paths:

- `newton/_src/solvers/semi_implicit/kernels_body.py::eval_body_joints()`
- `newton/_src/solvers/semi_implicit/kernels_body.py::joint_force()`
- `newton/_src/sim/builder.py::add_joint_ball()`
- PR #2304 review thread: https://github.com/newton-physics/newton/pull/2304#discussion_r3164780161

## Suggested acceptance criteria

- Add a regression test showing a ball joint with nonzero `joint_damping` changes SemiImplicit behavior compared with zero damping.
- Apply the damping term for all three ball-joint DOFs.
- Verify existing joint damping behavior for prismatic/revolute/D6 remains unchanged.

### PR DESCRIPTION
**Pull request by han-xudong**
## Description

Closes #2989.

- Apply `joint_damping` to all three angular DOFs in `SolverSemiImplicit` ball joints.
- Add a regression test showing SemiImplicit ball-joint angular velocity decays with nonzero passive damping.
- Update the unreleased changelog because this fixes user-facing solver behavior.

## Checklist

- [x] New or existing tests cover these changes
- [x] The documentation is up to date with these changes
- [x] `CHANGELOG.md` has been updated (if user-facing change)

## Test plan

```bash
uv run --extra dev -m newton.tests -k test_joint_damping
```

Result: 10 tests OK.

## Bug fix

Before this PR, `SolverSemiImplicit` ignored `Model.joint_damping` for `JointType.BALL` joints. Revolute, prismatic, and D6 paths already applied passive damping through their DOF force evaluation, but the ball-joint branch only applied `joint_f`.

**Steps to reproduce:**

1. Build a one-body articulation attached to world by a ball joint.
2. Set a nonzero initial angular velocity in `joint_qd`.
3. Compare `SolverSemiImplicit` behavior with `joint_damping=0.0` and `joint_damping>0.0`.
4. Observe that angular velocity does not decay without this PR.

**Minimal reproduction:**

```python
import newton

builder = newton.ModelBuilder(gravity=0.0)
body = builder.add_link(mass=1.0)
joint = builder.add_joint(
    newton.JointType.BALL,
    parent=-1,
    child=body,
    angular_axes=[
        newton.ModelBuilder.JointDofConfig(axis=newton.Axis.X, damping=3.0),
        newton.ModelBuilder.JointDofConfig(axis=newton.Axis.Y, damping=3.0),
        newton.ModelBuilder.JointDofConfig(axis=newton.Axis.Z, damping=3.0),
    ],
)
builder.add_articulation([joint])
builder.joint_qd[0:3] = [1.0, 0.0, 0.0]
model = builder.finalize()
solver = newton.solvers.SolverSemiImplicit(model, angular_damping=0.0)
```

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed ball-joint damping so it’s applied consistently across all three rotational degrees of freedom in semi-implicit solving.
* **Tests**
  * Added new automated coverage to confirm ball-joint angular velocity decays with damping enabled and stays constant when damping is zero, including updated device test wiring.

### CODE REVIEW

File: newton/tests/test_joint_damping.py [hunk 1]
```diff
@@ -87,6 +135,14 @@ def test_revolute_joint_damping_decays_velocity(test: TestJointDamping, device,
...
         )
 
+for device in devices:
+    add_function_test(
+        TestJointDamping,
+        "test_semi_implicit_ball_joint_damping_decays_velocity",
+        test_semi_implicit_ball_joint_damping_decays_velocity,
+        devices=[device],
+    )
+
```
[the remark this answers is not in the document]
Comment (han-xudong) [reply newton/tests/test_joint_damping.py:145]: The revolute damping test is still registered above this block: lines 127-136 iterate over `solvers` and register `test_revolute_joint_damping_decays_velocity_{solver_name}` for both Featherstone and SemiImplicit on each device. The loop at lines 138-144 only adds the new SemiImplicit ball-joint regression test, so applying the suggested change would duplicate SemiImplicit revolute coverage and would also bypass the existing `angular_damping=0.0` test setup. Verified locally with `uv run --extra dev -m newton.tests -k test_joint_damping` (10 tests passed).

**Review by eric-heiden — DISMISSED** [review]
Thanks @han-xudong, looks good to me! Could you fix the merge conflict in the changelog, then we can merge it.

**Comment by han-xudong** [comment]
Hi @eric-heiden, I rebased the branch onto the latest `main` and resolved the `CHANGELOG.md` merge conflict.

Local verification:
- `uvx pre-commit run -a`
- `uv run --extra dev -m newton.tests -k test_joint_damping` (10 tests passed)

This should be ready for re-review.

**Review by eric-heiden — APPROVED** [review]
Thanks!
